File: 20221012/모범답안/20221012_07/MinBetween.js
/**
 * @param arr {number[]}
 * @param a {number}
 * @param b {number}
 * @returns {number}
 */
function solution(arr, a, b) {
  let curMin = null
  arr.forEach(num => {
    const isIn = a <= num && num <= b
    const isMin = curMin === null || num < curMin
    curMin = isIn && isMin ? num : curMin
  })
  return curMin === null ? -1 : curMin
}

export default solution

File: 20221013/모범답안/20221013_07/MinBetween.py
def solution(arr, a, b):
    """
    :param arr: int[]
    :param a: int
    :param b: int
    :return: int
    """

    cur_min = None
    for num in arr:
        is_in = a <= num <= b
        is_min = cur_min is None or num < cur_min
        cur_min = num if is_in and is_min else cur_min

    return -1 if cur_min is None else cur_min
